### ISSUE TITLE
make build script for windows

### ISSUE THREAD

**Issue by brycefisher** [atoha/dbootstrap#29]
This is such a cool project. Sadly, I'm unable to get the build.py script to work on my WIndows Vista box with NodeJS / Stylus / Python 2.7.3 installed. 
### Steps to reproduce:

1) git clone recursively
2) cd dbootstrap
3) build.py demo
### Error:

```
INFO:dbootstrap.build:Building demo to path\to\dbootstrap\build\demo
INFO:dbootstrap.build:Building CSS.
Traceback (most recent call last):
File "path\to\dbootstrap\build.py", line 289, in <module>
    raise SystemExit(main())
  File "path\to\dbootstrap\build.py", line 66, in main
  os.path.join(dbootstrap_theme_path, 'index.styl')
File "path\to\dbootstrap\build.py", line 235, in execute
str(error)
__main__.CommandExecutionException: stylus 
--include path\to\dbootstrap\source \nib\lib 
--include path\to\dbootstrap\source\dbootstrap\theme\dbootstrap 
path\to\dbootstrap\source\dbootstrap\theme\dbootstrap\index.styl failed due to error: 
[Error 2] The system cannot find the file specified.
```

If I copy and paste the path, the index.styl file is in the correct place. Also, if I copy and paste the whole stylus command, that works as well. Silly windows....

**Comment by martinpengellyphillips** [comment]
Thanks for the report. Seems like the subprocess call is not able to find stylus. I don't have access to a Windows machine currently so might have to work on this blind for a bit.

**Comment by DavidSpriggs** [comment]
I had the same issue (building theme) and found that if I added shell=True to the subprocess.Popen calls I got past the stylus errors. Now there are node errors that seem to be path related:
![image](https://f.cloud.github.com/assets/661156/217357/4df7d788-84e7-11e2-9ce4-09f1fdd28a21.png)

If you run the node commands directly you get this:
![image](https://f.cloud.github.com/assets/661156/217370/9b456366-84e7-11e2-992f-7de0e3995ce4.png)

If you then change the paths to relative, it almost works:
![image](https://f.cloud.github.com/assets/661156/217377/f23e5d30-84e7-11e2-9c45-57b84caaa52a.png)

Any ideas? Also, can the built theme simply be posted to the repo or a branch?

**Comment by martinpengellyphillips** [comment]
Thanks for the extra information. I'll try and grab a Windows machine to lock this down soon. In the meantime, have followed your suggestion and uploaded a tar of the built theme directory structure to http://thesociable.github.com/dbootstrap/download/dbootstrap.tar.gz  Try downloading and decompressing that to see if it works for you until the Windows build issue is sorted.

**Comment by hdd** [comment]
I don't know if it's related , but this might be useful : http://stackoverflow.com/questions/13231168/how-to-do-a-dojo-build-using-nodejs

**Comment by martinpengellyphillips** [comment]
The issue is similar, but in this case I suspect it is actually Python subprocess that cannot find nodejs in the env.

**Comment by martinpengellyphillips** [comment]
Finally got hold of a windows machine so can dig into this properly.

**Comment by martinpengellyphillips** [comment]
Ok, have pushed some fixes for a few of the errors. Now looking into the Node build issue referenced at https://bugs.dojotoolkit.org/ticket/15413

**Comment by martinpengellyphillips** [comment]
Merged in #51
